java/src/main/java/ai/rapids/cudf/ParquetWriterOptions.java

```diff
@@ -39,32 +45,119 @@
+
+    /**
+     * Add a metadata key and a value
+     * @param key
+     * @param value
+     */
+    public Builder withMetadata(String key, String value) {
+      this.metadata.put(key, value);
+      return this;
+    }
+
+    /**
+     * Add a map of metadata keys and values
+     * @param metadata
+     */
+    public Builder withMetadata(Map<String, String> metadata) {
+      this.metadata.putAll(metadata);
+      return this;
+    }
+
+    /**
+     * Set the compression type to use for writing
+     * @param compression
+     */
+    public Builder withCompressionType(CompressionType compression) {
+      this.compressionType = compression;
+      return this;
+    }
 
     public Builder withStatisticsFrequency(StatisticsFrequency statsGranularity) {
       this.statsGranularity = statsGranularity;
       return this;
     }
 
     /**
-     * Set whether the timestamps should be written in INT96
+     * Set column name
+     * @param name
+     */
+    public Builder withColumn(String... name) {
+      IntStream.range(0, name.length).forEach(
+          i -> columnOptions.add(ParquetColumnWriterOptions.builder()
+                  .withColumnName(name[i])
+                  .build())
+      );
+      return this;
+    }
+
+    /**
+     * Set the decimal precision
+     * @param name
+     * @param precision
+     */
+    public Builder withDecimalColumn(String name, int precision) {
+      columnOptions.add(ParquetColumnWriterOptions.builder()
+          .withColumnName(name)
+          .withDecimalPrecision(precision).build());
+      return this;
+    }
+
+    /**
+     * Set the nullable decimal precision
+     * @param name
+     * @param precision
+     */
+    public Builder withNullableDecimalColumn(String name, int precision) {
+      columnOptions.add(ParquetColumnWriterOptions.builder()
+          .withColumnName(name)
+          .withNullable(true)
+          .withDecimalPrecision(precision).build());
+      return this;
+    }
+
+    /**
+     * Set nullable column name
+     * @param name
      */
-    public Builder withTimestampInt96(boolean int96) {
-      this.isTimestampTypeInt96 = int96;
+    public Builder withNullableColumn(String... name) {
+      IntStream.range(0, name.length).forEach(
+          i -> columnOptions.add(ParquetColumnWriterOptions.builder()
+              .withNullable(true)
+              .withColumnName(name[i])
+              .build())
+      );
       return this;
     }
 
     /**
-     * This is a temporary hack to make things work.  This API will go away once we can update the
-     * parquet APIs properly.
-     * @param precisionValues a value for each column, non-decimal columns are ignored.
-     * @return this for chaining.
+     * Set a struct column
+     * @param options
      */
-    public Builder withDecimalPrecisions(int ... precisionValues) {
-      this.precisionValues = precisionValues;
+    public Builder withStructColumn(ParquetColumnWriterOptions options) {
```

Comment:
This has nothing to do with structs, and there is nothing for an list column.  Could we just have it be `withColumn`?

Reply:
done. PTAL